hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -204,16 +204,16 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 
 	// Reconcile the shared provider credentials secret by resolving the reference
 	// from the HostedCluster and syncing the secret in the control plane namespace.
-	var hostedClusterProviderCredsSecret corev1.Secret
-	err = r.Client.Get(ctx, client.ObjectKey{Namespace: hcluster.GetNamespace(), Name: hcluster.Spec.ProviderCreds.Name}, &hostedClusterProviderCredsSecret)
+	var cloudControllerCreds corev1.Secret
+	err = r.Client.Get(ctx, client.ObjectKey{Namespace: hcluster.GetNamespace(), Name: hcluster.Spec.Platform.AWS.KubeCloudControllerCreds.Name}, &cloudControllerCreds)
```

Comment:
Should add a check for Spec.Platform.AWS != nil

Reply:
Thanks, it also now occurs to me we're missing the discriminating field (PlatformType) on our discriminating union type (PlatformSpec), I'll add that too